bson/src/main/org/bson/internal/CodecRegistryHelper.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.internal;
+
+import org.bson.UuidRepresentation;
+import org.bson.codecs.configuration.CodecConfigurationException;
+import org.bson.codecs.configuration.CodecProvider;
+import org.bson.codecs.configuration.CodecRegistry;
+
+public final class CodecRegistryHelper {
+
+    public static CodecRegistry createRegistry(final CodecRegistry codecRegistry, final UuidRepresentation uuidRepresentation) {
+        CodecRegistry retVal = codecRegistry;
+        if (uuidRepresentation != UuidRepresentation.JAVA_LEGACY) {
```

Comment:
So a user cannot default to a custom `uuidRepresentation` in settings and then set an alternative Codec Registry?

I was concerned that would be an issue but as most CodecRegistries will be created via the fromCodecs, fromProviders helpers I think users shouldn't hit this often.

Reply:
They can, but if they want the registry to ignore the `uuidRepresentation` they have to use a codec that doesn't implement `OverridableUuidRepresentationCodec`, e.g. `UuidCodec` instead of `OverridableUuidRepresentationUuidCodec`.

So an existing application that already overrides the default representation by setting a registry with their own `UuidCodec` will not be affected by the new `uuidRepresentation` setting.

I might be misunderstanding your question though.